### PR TITLE
Implement -terse option

### LINKED ISSUE #22: Allow to use -terse mode of qalc
Nice project!

I'd like `qalc` instance to run with `-t` / `-terse` option, not only it gives, well, terse output, but also it removes the annoying parsing errors while I'm typing my expression.

Compare the current master:

before

With a patched `rofi-calc` that adds `-t`:

after

I'd like to avoid patching the sources 😉 

The request is therefore to consider using `-t` as a default, or providing a configuration for it, maybe make the entire `qalc` command with all of its parameters configurable.

### PR DESCRIPTION
**Pull request by maximbaz**
Fixes #22 

I have a suspicion that what I wrote can be done in a fewer amount of lines, please feel free to make suggestions 😉 But at least it works 😄

### CODE REVIEW

**Comment by svenstaro** [comment]
Also, I think you can do this much more simply using https://developer.gnome.org/glib/stable/glib-Arrays.html instead of the static array. Don't forget to free the memory afterward. :)

**Comment by maximbaz** [comment]
I tried to make use of GArray, but unless I'm missing something it's not making code any simpler, in the end I still need a variable of type `const gchar**` in order to feed it to `g_subprocess_newv()`, so it would require an extra loop for conversion. Am I missing something?

I did find a way to simply a bit, and now I'm also freeing memory 😄 

Please have another look 😉

**Comment by svenstaro** [comment]
First of all, I'd really prefer to not to have any manual `malloc`s in this code if it can be helped. Secondly, if you have the `GArray`, you should be able to cast it to a `const gchar**` as it should have the same memory layout.

File: src/calc.c [hunk 1]
```diff
@@ -310,8 +316,23 @@ static char* calc_preprocess_input(Mode* sw, const char* input)
     GError *error = NULL;
     CALCModePrivateData* pd = (CALCModePrivateData*)mode_get_private_data(sw);
 
-    const gchar* const argv[] = { "qalc", "+u8", "-s", "update_exchange_rates 1days", input, NULL };
+    const gchar* const default_args[] = { "+u8", "-s", "update_exchange_rates 1days", input, NULL };
+    int len = sizeof(default_args) / sizeof(const gchar*);
```
Comment (svenstaro) [inline src/calc.c]: You can use [G_N_ELEMENTS()](https://developer.gnome.org/glib/stable/glib-Standard-Macros.html#G-N-ELEMENTS:CAPS) here instead.

File: src/calc.c [hunk 2]
```diff
@@ -310,8 +316,23 @@ static char* calc_preprocess_input(Mode* sw, const char* input)
...
+
+    int n_extra_args = 2; // "qalc" and optional  "-t"
+    const gchar** argv = malloc((len + n_extra_args) * sizeof(const gchar*));
+
+    int i = 0;
+    argv[i++] = "qalc";
+    if (find_arg(TERSE_OPTION) > -1) {
+        argv[i++] = "-t";
+    }
+    for (int j = 0; j < len; j++) {
```
Comment (svenstaro) [inline src/calc.c]: The reason I think using `GArray` would be nice here is so you can get read of this for loop as you can just insert directly into the old array.

**Comment by maximbaz** [comment]
I think GArray is implemented as a linked list internally, so it can't have the same memory layout as an array. I'll push a commit now, is that what you expected to see? It segfaults for me...

https://developer.gnome.org/glib/stable/glib-Arrays.html#glib-Arrays.other_details

**Comment by svenstaro** [comment]
I'm pretty sure it's not a linked list as 1) they explicitly have another data structure which is a linked list 2) they state that it's similar to C-arrays and 3) they state that the memory behind `gchar *data` may move which doesn't happen with a linked list.

I think you have to access the `.data` member of the `GArray` and use that. That will probably be a C-compatible structure. From the looks of it, you're currently putting the whole `GArray` in there which won't work as it has the `len` in the struct as well but the subprocess expects a `NULL` terminated list.

**Comment by svenstaro** [comment]
Looks like smart C-array manipulation to me so we should be good: https://github.com/GNOME/glib/blob/master/glib/garray.c

**Comment by svenstaro** [comment]
Great work!

**Pull request merged**